x/auth/client/cli/sign.go

```diff
@@ -122,35 +125,58 @@ func makeSignCmd(cdc *amino.Codec) func(cmd *cobra.Command, args []string) error
 		if err != nil {
 			return err
 		}
+
 		defer fp.Close()
 		fmt.Fprintf(fp, "%s\n", json)
+
 		return
 	}
 }
 
-func printSignatures(stdTx auth.StdTx) bool {
+func printAndValidateSigs(cliCtx context.CLIContext, chainID string, stdTx auth.StdTx) bool {
 	fmt.Println("Signers:")
+
 	signers := stdTx.GetSigners()
 	for i, signer := range signers {
 		fmt.Printf(" %v: %v\n", i, signer.String())
 	}
 
+	success := true
```

Comment:
probably a better way which we can avoid creating/using this variable altogether (for another PR though)

Reply:
@rigelrozanski I'm happy to do that in this PR since it's really in context. Do you have suggestions though? 

We prefer to loop over all the sigs and _not_ short-circuit. In other words, we cannot simply return an error or bool at first sight of a problem. Any suggestions on how to do this without keeping track via a variable?

Reply:
After speaking with @alessio we've decided to short circuit and thus we can avoid the use of a tracking variable.

Reply:
nope, if the point is to complete the loop before returning an error, then a variable must be used.

Reply:
The idea is that txs can have many signers/signatures. The UX we want to provide should show you all the signatures and either an OK or the ERROR. If we short-circuit, the user doesn't get the full picture.

I'd propose the function is ok.